iceoryx_utils/include/iceoryx_utils/internal/units/duration.inl

```diff
@@ -21,47 +21,102 @@ namespace iox
 namespace units
 {
 template <typename T>
-constexpr Duration Duration::nanoseconds(const T ns)
+constexpr Duration Duration::nanoseconds(const T value)
 {
-    return operator"" _ns(ns);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
+    }
+    return operator"" _ns(value);
```

Comment:
Too much code,
```cpp
return (value < 0 ) ? Duration(0U, 0U) : operator""_ns(value);
```

Would spare us of the four additional lines. Could you please change that for this and all the other methods.

Reply:
there is now a print to inform the user of the clamping, therefore your proposal cannot be used anymore